File: 206-Reverse-Linked-List/v2.py
from typing import Optional

# Definition for singly-linked list.
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next


class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if head is None:
            return None
        result_head = None
        while head is not None:
            result_head = ListNode(head.val, result_head)
            head = head.next

        return result_head

File: 2-Add-Two-Numbers/v2.java
class Solution {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
        ListNode dummy = new ListNode();
        ListNode cur = dummy;
        int total = 0;
        
        while (l1 != null || l2 != null || total > 0) {
            if (l1 != null) {
                total += l1.val;
                l1 = l1.next;
            }
            if (l2 != null) {
                total += l2.val;
                l2 = l2.next;
            }

            cur.next = new ListNode(total % 10);
            cur = cur.next;
            total = total / 10;
        }

        return dummy.next;
    }
}
